File: Homework/Duc_Trinh/75-sort-color.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
  let nextCorrectPosToSet = 0;
  // set 0s in correct place
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === 0) {
      const temp = nums[nextCorrectPosToSet];
      nums[nextCorrectPosToSet] = nums[i];
      nums[i] = temp;
      nextCorrectPosToSet = nextCorrectPosToSet + 1;
    }
  }
  // set 1s in correct place
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === 1) {
      const temp = nums[nextCorrectPosToSet];
      nums[nextCorrectPosToSet] = nums[i];
      nums[i] = temp;
      nextCorrectPosToSet = nextCorrectPosToSet + 1;
    }
  }
};

File: Homework/Ngan_Luu/75-sort-color.py
#https://leetcode.com/problems/sort-colors/
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        for a in [0,1,2]:
            p=0
            for i in range(1,len(nums)):
                if nums[p]!=a:
                    p+=1
                elif nums[i]!=a:
                    nums[i],nums[p]=nums[p],nums[i]
                    p+=1